crates/rover-client/src/operations/graph/publish/types.rs

```diff
@@ -33,8 +37,100 @@ impl From<GitContext> for GraphPublishContextInput {
     }
 }
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Serialize, Debug, PartialEq)]
 pub struct GraphPublishResponse {
-    pub schema_hash: String,
-    pub change_summary: String,
+    pub api_schema_hash: String,
+    #[serde(flatten)]
+    pub change_summary: ChangeSummary,
+}
+
+#[derive(Clone, Serialize, Debug, PartialEq)]
+pub struct ChangeSummary {
+    pub field_changes: FieldChanges,
+    pub type_changes: TypeChanges,
+}
+
+impl ChangeSummary {
+    pub(crate) fn none() -> ChangeSummary {
+        ChangeSummary {
+            field_changes: FieldChanges::none(),
+            type_changes: TypeChanges::none(),
+        }
+    }
+
+    pub(crate) fn is_none(&self) -> bool {
+        self.field_changes.is_none() && self.type_changes.is_none()
+    }
+}
+
+impl fmt::Display for ChangeSummary {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.is_none() {
+            write!(f, "[No Changes]")
+        } else {
+            write!(f, "[{}, {}]", &self.field_changes, &self.type_changes)
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Debug, PartialEq)]
+pub struct FieldChanges {
+    pub additions: u64,
+    pub removals: u64,
+    pub edits: u64,
+}
+
+impl FieldChanges {
+    pub(crate) fn none() -> FieldChanges {
```

Comment:
As an addition to the comment on having `ChangeSummary::with_diff`, you can also implement `FieldChanges::with_diff` that `ChangeSummary::with_diff` can call under the hood.